rubicon/objc/api.py

```diff
@@ -851,17 +827,39 @@ def __setattr__(self, name, value):
                 if isinstance(value, enum.Enum):
                     value = value.value
                 ObjCBoundMethod(method, self)(value)
-            else:
-                # Add a new Python attribute.
-                # These attributes are currently only stored on the Python ObjCInstance object
-                # and not on the actual Objective-C object.
-                # To ensure that the attributes aren't lost when the ObjCInstance has no more Python references
-                # (but the Objective-C object still has Objective-C references
-                # and may be wrapped in an ObjCInstance again in the future),
-                # every ObjCInstance with additional Python attributes is "kept alive" on the Python side.
-                # This prevents the ObjCInstance from being destroyed while the Objective-C object still exists.
-                self._ensure_keep_alive()
+            elif name.startswith("_"):
```

Comment:
I did run into trouble with the `_needs_release` attribute previously but cannot reproduce it now. As you say, it will already exist in the `__dict__` and is therefore caught by the previous check. Pending any upcoming issues, I'll remove the special handling of private attributes.